File: An 2/Sem 1/MAP/Lab7/C#/Lab2/Model/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2
{
	namespace Lab2_Model {
		class Student : Comparable<Student>
	    {
	        public string name;
	        public int id;
	        public int grade;

	        public Student(int id, string name, int grade)
	        {
	            this.id = id;
	            this.name = name;
	            this.grade = grade;
	        }

			public virtual float average() {
				return grade;
			}

			bool Comparable<Student>.isGreaterThan(Student student) {
				return (this.grade > student.grade);
			}

			public override String ToString() {
				return String.Format("{0}|{1}|{2}|", this.id, this.name, this.grade);
			}
	    }
	}
}

File: An 2/Sem 2/AOP/Lab 2/src/com/company/Model/Student.java
package com.company.Model;

import java.util.Map;
import java.util.logging.Logger;

/**
 *
 * @author mihai
 */
public class Student implements Comparable<Student>, Readable, HasId, Buildable {
    public String name;
    public int id;
    public int grade;
    private static Logger logger = Logger.getLogger("Students");

    public Student() {

    }

    public Student(int id, String name, int grade){
        logger.info("[Entering:]Student.init");
        this.name = name;
        this.id = id;
        this.grade = grade;
    }

    public float average() {
        logger.info("[Entering:]Student.average");
        return grade;
    }

    @Override
    public boolean isGreaterThan(Student student) {
        logger.info("[Entering:]Student.isGreaterThan");
        return (this.grade > student.grade);
    }

    @Override
    public String toString() {
        logger.info("[Entering:]Student.toString");
        return String.format("%s|%d|%s|%d|", this.getClass().toString(), this.id, this.name, this.grade);
    }

    @Override
    public void readAttributesFromString(String string) {
        logger.info("[Entering:]Student.readAttributesFromString");
        String[] tokens = string.split("[|]+");
        this.id = Integer.parseInt(tokens[1]);
        this.name = tokens[2];
        this.grade = Integer.parseInt(tokens[3]);
    }

    @Override
    public Integer getId() {
        logger.info("[Entering:]Student.getId");
        return this.id;
    }

    @Override
    public void buildObjectFromMap(Map<String, String> map) {
        logger.info("[Entering:]Student.buildObjectFromMap");
        this.id = Integer.parseInt(map.get("id"));
        this.name = map.get("name");
        this.grade = Integer.parseInt(map.get("grade"));
    }
}
